src/prefect/cli/worker.py

```diff
@@ -156,5 +159,16 @@ async def start(
 
             started_event = await worker._emit_worker_started_event()
 
+            # start healthcheck ASGI server
+            # in a background thread
+            server_thread = threading.Thread(
+                name="healthcheck-server-thread",
+                target=partial(
+                    start_healthcheck_server, worker=worker, run_once=run_once
+                ),
+                daemon=True,
+            )
+            server_thread.start()
```

Comment:
so threading this server allows us to propagate a SIGINT and gracefully kill this portion when the worker dies

```
$ prefect worker start -p eddie-pool
Discovered worker type 'process' for work pool 'eddie-pool'.
Worker 'ProcessWorker 5302d79a-d2ae-4cb8-9852-6f071c242fd0' started!
INFO:     Started server process [60317]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:     127.0.0.1:53575 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:53576 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:53577 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:53578 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:53579 - "GET /health HTTP/1.1" 200 OK
^CReceived SIGINT. Sending SIGINT to the process worker (PID 60317)...
Worker 'ProcessWorker 5302d79a-d2ae-4cb8-9852-6f071c242fd0' stopped!

Aborted.

```

Reply:
and those health check logs correspond to something like this:

```
$ curl -i localhost:8080/health
HTTP/1.1 200 OK
date: Thu, 25 May 2023 21:37:12 GMT
server: uvicorn
content-length: 112
content-type: application/json

{"name":"ProcessWorker 5302d79a-d2ae-4cb8-9852-6f071c242fd0","worker_pool":"eddie-pool","worker_pool_queues":[]}%
```